spec/controllers/sign_up/registrations_controller_spec.rb

```diff
@@ -56,6 +56,36 @@
         )
       end
     end
+
+    context 'with thmx enabled' do
+      let(:tmx_session_id) { '1234' }
+
+      before do
+        allow(FeatureManagement).to receive(:account_creation_device_profiling_collecting_enabled?).
+          and_return(true)
+        allow(IdentityConfig.store).to receive(:lexisnexis_threatmetrix_org_id).and_return('org1')
+        allow(IdentityConfig.store).to receive(:lexisnexis_threatmetrix_mock_enabled).
+          and_return(false)
+        subject.session[:threatmetrix_session_id] = tmx_session_id
+      end
+
+      it 'renders new valid request' do
+        tmx_url = 'https://h.online-metrix.net/fp'
+        expect(subject).to receive(:render).with(
```

Comment:
I wish there was a way to test the assigned locals without duplicating so much of the implementation itself, but everything I find seems to point to the same approach 🤷

Reply:
Yea. Its really frustrating.